src/h3lib/lib/faceijk.c

```diff
@@ -934,3 +924,29 @@ Overage _adjustPentVertOverage(FaceIJK *fijk, int res) {
     } while (overage == NEW_FACE);
     return overage;
 }
+
+/**
+ * Encodes a coordinate on the sphere to the corresponding icosahedral face and
+ * containing the squared euclidean distance to that face center.
+ *
+ * @param g The spherical coordinates to encode.
+ * @param face The icosahedral face containing the spherical coordinates.
+ * @param sqd The squared euclidean distance to its icosahedral face center.
+ */
+void _geoToClosestFace(const LatLng *g, int *face, double *sqd) {
+    Vec3d v3d;
+    _geoToVec3d(g, &v3d);
+
+    // determine the icosahedron face
+    *face = 0;
+    // The distance between two farthest points is 2.0, therefore the square of
+    // the distance between two points should always be less or equal than 4.0 .
+    *sqd = 5.0;
+    for (int f = 0; f < NUM_ICOSA_FACES; ++f) {
```

Comment:
why not set sqd to `_pointSquareDist(&faceCenterPoint[0], &v3d)` and start `f` from 1, as before?

Reply:
This is a bit of a wash IMO - on the one hand you have a duplicated function call, on the other you have specialized knowledge/assumptions about the output of `_pointSquareDist`. I think either is fine.